back_end/src/main/java/com/software/grey/repositories/PostRepository.java

```diff
@@ -16,30 +16,43 @@
 @Repository
 public interface PostRepository extends JpaRepository<Post, String> {
 
-
     Post findByUser(User user);
 
-    /*
-        To select posts that the user wrote and filter them by day, month and year and sort them descendingly.
+    /**
+     * To select posts that the user wrote and filter them by day, month and year and sort them descendingly.
+     *
+     * @param userName the username of the user
+     * @param day      the day of the month to filter with
+     * @param month    the month of the year to filter with
+     * @param year     the year to filter with
+     * @param pageable the pagination information
+     * @return the page of posts the user wrote in the specified day, month and year
      */
     @Query("""
-        SELECT p FROM Post p
-        WHERE p.user.username = :userName
-        AND (:day IS NULL OR DAY(p.postTime) = :day)
-        AND (:month IS NULL OR MONTH(p.postTime) = :month)
-        AND (:year IS NULL OR YEAR(p.postTime) = :year)
-        """)
-    Page<Post> findDiaryByUsernameAndDayMonthYear(
+            SELECT p FROM Post p
+            WHERE p.user.username = :userName
+                AND (:day IS NULL OR DAY(p.postTime) = :day)
+                AND (:month IS NULL OR MONTH(p.postTime) = :month)
+                AND (:year IS NULL OR YEAR(p.postTime) = :year)
+            ORDER BY p.postTime DESC
+            """)
+    Page<Post> findDiaryByUsernameAndDayMonthYearSortedByDate(
             @Param("userName") String userName,
             @Param("day") Integer day,
             @Param("month") Integer month,
             @Param("year") Integer year,
             Pageable pageable
     );
 
-    // This query returns the feelings the user wrote about in their last 5 posts, and their frequency
-    // the inner query returns the feelings in the user's last 5 posts
-    // the outer query sums those feelings and returns their frequency
+
+    /**
+     * This query returns the feelings the user wrote about in their last 5 posts, and their frequency the inner
+     * query returns the feelings in the user's last 5 posts the outer query sums those feelings and returns
+     * their frequency
```

Comment:
For better readability, start a new line beginning from ``` the inner query ...``` and another new line from ``` the outer query```. Just like what was written before.